src/webview/components/memory-table.tsx

```diff
@@ -220,33 +272,76 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
             tableStyle: { minWidth: '30rem' },
             value: rows
         };
+    }
+
+    protected createScrollingBehaviorSpecificProperties(props: DataTableProps<MemoryRowData[]>): DataTableProps<MemoryRowData[]> {
+        if (this.props.scrollingBehavior === 'Auto-Append') {
+            return {
+                ...props,
+                pt: {
+                    wrapper: {
+                        onScroll: event => {
+                            if (!this.isLoading) {
+                                // Append new data only if we reach the bottom
+                                const element = event.target as HTMLDivElement;
+                                const distanceBottom = Math.abs(element.scrollHeight - element.scrollTop - element.clientHeight);
+
+                                if (distanceBottom < 1) {
+                                    this.appendMoreMemory();
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+        } else {
+            return {
+                ...props,
+                footer: this.renderFooter()
+            };
+        }
+    }
 
+    /**
+     * This method asserts that we have sufficient rows visible to enable vertical scrollbars
+     */
+    protected ensureSufficientVisibleRowsForScrollbar(): void {
+        const datatableValues = this.datatableRef.current?.props.value;
+        if (datatableValues && datatableValues.length < this.rowsToAppendCount) {
+            // We have too few rows, we need to load more data
+            this.appendMoreMemory();
+        }
+    }
+
+    protected appendMoreMemory(): void {
+        if (!this.isLoading && MemoryState.hasMemory(this.props)) {
+            const options = this.createMemoryRowListOptions(this.props.memory, this.props);
+            const newCount = this.props.memoryReadArguments.count + options.wordsPerRow * this.rowsToAppendCount;
+            this.props.fetchMemory({ count: newCount });
+        }
     }
 
     protected onSelectionChanged = (event: DataTableSelectionCellChangeEvent<MemoryRowData[]>) => {
         this.setState(prev => ({ ...prev, selection: event.value }));
     };
 
     protected renderHeader(): React.ReactNode | undefined {
-        const { offset, count, fetchMemory, scrollingBehavior } = this.props;
-
         let memorySelect: React.ReactNode | undefined;
         let loading: React.ReactNode | undefined;
 
-        if (this.isShowMoreEnabled) {
+        if (this.isShowMoreEnabled && MemoryState.hasMemory(this.props)) {
             memorySelect = <div className='flex-auto'>
                 <MoreMemorySelect
-                    offset={offset}
-                    count={count}
+                    memoryReadArguments={this.props.memoryReadArguments}
```

Comment:
Separately: should we show the manual more memory select if the user has opted into automatic memory fetching?

Reply:
The only disadvantage would be that the users would try to click on the button instead of scrolling, but I'm not against it.

Reply:
> Separately: should we show the manual more memory select if the user has opted into automatic memory fetching?

What use case do you have in mind with the button? One advantage I see is that a user could pre-load memory before starting to scroll. But I am also a little concerned about @haydar-metin 's point. Probably would need to try it out to see whether it's useful or if it feels off.

Reply:
Oh, my apologies. Misunderstood the above discussion. I don't necessarily see the pre-load functionality as something I'd expect many to use. The use cases I have in mind are pretty much all around just scrolling without pressing a button to load more.